nomenclature/codelist.py

```diff
@@ -280,34 +280,70 @@ def to_yaml(self, path=None):
         else:
             return stream
 
-    def to_excel(self, excel_writer, sheet_name="definitions"):
-        """Write the codelist to an Excel spreadsheet
+    def to_pandas(self, sorted=False):
+        """Export the CodeList to a :class:`pandas.DataFrame`
 
         Parameters
         ----------
-        excel_writer : str, path or ExcelWriter object
-            Any valid string path, :class:`pathlib.Path` or :class:`pandas.ExcelWriter`
-        sheet_name : str
-            Name of sheet that will contain the codelist
+        sorted : bool, optional
+            Sort the codelist before exporting to csv.
         """
-
-        # open a new ExcelWriter instance (if necessary)
-        close = False
-        if not isinstance(excel_writer, pd.ExcelWriter):
-            close = True
-            excel_writer = pd.ExcelWriter(excel_writer)
-
-        # format and write the codelist
         codelist = (
             pd.DataFrame.from_dict(self.mapping, orient="index")
             .reset_index()
             .rename(columns={"index": self.name})
             .drop(columns="file")
         )
+        if sorted:
+            codelist.sort_values(by=self.name, inplace=True)
         codelist.rename(
             columns={c: str(c).capitalize() for c in codelist.columns}, inplace=True
         )
-        write_sheet(excel_writer, sheet_name, codelist)
+        return codelist
+
+    def to_csv(self, path=None, sort_by_code=False, **kwargs):
+        """Write the codelist to a comma-separated values (csv) file
+
+        Parameters
+        ----------
+        path : str, path or file-like, optional
+            File path as string or :class:`pathlib.Path`, or file-like object.
+            If *None*, the result is returned as a csv-formatted string.
+            See :meth:`pandas.DataFrame.to_csv` for details.
+        sort_by_code : bool, optional
+            Sort the codelist before exporting to csv.
+        **kwargs
+            Passed to :meth:`pandas.DataFrame.to_csv`.
+        """
+        return self.to_pandas(sort_by_code).to_csv(path, **kwargs)
+
+    def to_excel(self, excel_writer, sheet_name=None, sort_by_code=False, **kwargs):
```

Comment:
Maybe this could be reduced a bit more to just take a path and write out the codelist similar to `to_csv`. I'm not sure we need the option to pass and ExcelWriter object and the call to pyam to write the sheet.
But if you want all the fine grained control that's totally fine as well.

Reply:
First, my philosophy here is to follow pyam, and pyam follows pandas - in the spirit of "least surprise".

Also, if you want to write several codelists from one **DataStructureDefintion** to an Excel file, you could do:

```python
writer = pd.ExcelWriter()
for dim in dsd.dimensions:
    dsd[dim].to_excel(writer)
writer.close()
```

Thanks to your suggestion, I realized that **DataStructureDefintion** also has a `to_excel()` method - I've now streamlined that and use the **CodeList** method directly.

Reply:
> First, my philosophy here is to follow pyam, and pyam follows pandas - in the spirit of "least surprise".

Ah yes, good point, keeping the APIs aligned is always a good idea. 

> Thanks to your suggestion, I realized that DataStructureDefintion also has a to_excel() method - I've now streamlined that and use the CodeList method directly.

That's in a different PR though, right? I just checked and I could not find any changes to `definitions.py`.

Reply:
Forgot to push the commit, sorry.